flyteplugins/go/tasks/pluginmachinery/internal/webapi/core.go

```diff
@@ -65,25 +68,91 @@ func (c CorePlugin) GetProperties() core.PluginProperties {
 	return core.PluginProperties{}
 }
 
+func (c CorePlugin) getPlugin(pluginType string) (webapi.AsyncPlugin, webapi.SyncPlugin, error) {
```

Comment:
I would prefer having two functions: toSyncPlugin and toAsyncPlugin. The call site knows which to call already so it seems weird passing a string to this function to control what to cast to. This function has to either return sync or async interface, which is also a bit strange.

Reply:
No problem, I think this is better!
I will do it now

Reply:
Thanks really much

Reply:
Done, I also split the `Handle` function into `syncHandle` and `asyncHandle`, which is more clean and readable.
Thanks for you and Kevin's review!